crates/catalog/rest/src/catalog.rs

```diff
@@ -504,8 +504,15 @@ impl Catalog for RestCatalog {
             .query::<LoadTableResponse, ErrorResponse, OK>(request)
             .await?;
 
+        let config = resp
+            .config
+            .unwrap_or_default()
+            .into_iter()
+            .chain(self.user_config.props.clone().into_iter())
```

Comment:
But if it was that way around, you could not use the local user_config to override a property that came from the REST service? I think we would want the ability to do that. I think it makes more sense the way around that I have it.

Reply:
My rationale behind this was so that users _could_ override values from the REST config. For instance if you wanted to try out using an S3 proxy / cache by overriding the value of the S3 warehouse endpoint. I can change it if you insist, but that would reduce the usefulness of this change in my eyes.

Reply:
I don't have a strong preference for this but respecting the user's override seems sensible to me. We can mention its behavior in the API comments.

Reply:
I've added some doc comments to the two relevant methods to clarify the behaviour. Hope this helps.